cmake/Gems.cmake

```diff
@@ -252,3 +252,68 @@ function(ly_enable_gems_delayed)
         endforeach()
     endforeach()
 endfunction()
+
+#! ly_is_gem_available: Check if a gem is available in the current project configuration.
+# \arg:NAME name of the gem to check for availability.
+# \arg:OUTPUT_VARIABLE name of the variable in which to set the returned value (either TRUE or FALSE).
+# \arg:VARIANT name of the project variant if which to check for availability. Defaults to 'Clients'.
+function(ly_is_gem_available)
+    set(options)
+    set(oneValueArgs NAME OUTPUT_VARIABLE VALUE_IF_FOUND VALUE_IF_NOT_FOUND VARIANT)
```

Comment:
`VALUE_IF_FOUND` and `VALUE_IF_NOT_FOUND` can be removed since we're using `OUTPUT_VARIABLE`